server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -19,81 +19,342 @@
+        preference = in.readOptionalString();
+        originalIndices = OriginalIndices.readOriginalIndices(in);
+    }
 
-    void setAliasFilter(AliasFilter filter);
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        innerWriteTo(out, false);
+        OriginalIndices.writeOriginalIndices(originalIndices, out);
+    }
 
-    void source(SearchSourceBuilder source);
+    protected final void innerWriteTo(StreamOutput out, boolean asKey) throws IOException {
+        shardId.writeTo(out);
+        out.writeByte(searchType.id());
+        if (!asKey) {
+            out.writeVInt(numberOfShards);
+        }
+        out.writeOptionalWriteable(scroll);
+        out.writeOptionalWriteable(source);
+        if (out.getVersion().before(Version.V_8_0_0)) {
+            // types not supported so send an empty array to previous versions
+            out.writeStringArray(Strings.EMPTY_ARRAY);
+        }
+        aliasFilter.writeTo(out);
+        out.writeFloat(indexBoost);
+        if (asKey == false) {
+            out.writeVLong(nowInMillis);
+        }
+        out.writeOptionalBoolean(requestCache);
+        out.writeOptionalString(clusterAlias);
+        out.writeBoolean(allowPartialSearchResults);
+        if (asKey == false) {
+            out.writeStringArray(indexRoutings);
+            out.writeOptionalString(preference);
+        }
+    }
 
-    int numberOfShards();
+    @Override
+    public String[] indices() {
+        if (originalIndices == null) {
+            return null;
+        }
+        return originalIndices.indices();
+    }
 
-    SearchType searchType();
+    @Override
+    public IndicesOptions indicesOptions() {
+        if (originalIndices == null) {
+            return null;
+        }
+        return originalIndices.indicesOptions();
+    }
 
-    float indexBoost();
+    public ShardId shardId() {
+        return shardId;
+    }
 
-    long nowInMillis();
+    public SearchSourceBuilder source() {
+        return source;
+    }
 
-    Boolean requestCache();
+    public AliasFilter getAliasFilter() {
+        return aliasFilter;
+    }
 
-    boolean allowPartialSearchResults();
+    public void setAliasFilter(AliasFilter aliasFilter) {
+        this.aliasFilter = aliasFilter;
+    }
 
-    Scroll scroll();
+    public void source(SearchSourceBuilder source) {
+        this.source = source;
+    }
 
-    /**
-     * Returns the routing values resolved by the coordinating node for the index pointed by {@link #shardId()}.
-     */
-    String[] indexRoutings();
+    public int numberOfShards() {
+        return numberOfShards;
+    }
 
-    /**
-     * Returns the preference of the original {@link SearchRequest#preference()}.
-     */
-    String preference();
+    public SearchType searchType() {
+        return searchType;
+    }
 
-    /**
-     * Returns the cache key for this shard search request, based on its content
-     */
```

Comment:
Maybe we could keep the javadocs that was there although we have a single impl?

Reply:
good catch, thanks